tools/lisa-buildroot-update-kernel-config

```diff
@@ -51,8 +51,26 @@ if [[ -z "$1" || -z "$2" ]]; then
 	exit -1
 fi
 
-set -u
+# Automatically detect the arm arch we are building for
+export ARCH=${ARCH:-$(grep -owE "(arm|arm64)" "$KERNEL_CONFIG")}
+if [ -z "$ARCH" ]; then
+	echo "ERROR: set ARCH environment variable"
+	exit -1
+fi
+
+echo "Assuming ARCH=$ARCH"
+
+echo "Merging Kconfig fragment into $KERNEL_CONFIG"
+fragment=$(mktemp)
 
 # Update kernel .config to use the provided rootfs as builtin initramfs
-sed -i "s#[.]*CONFIG_INITRAMFS_SOURCE.*#CONFIG_INITRAMFS_SOURCE=\"$ROOTFS\"#" "$KERNEL_CONFIG"
-make -C "$KERNEL_DIR" olddefconfig
+# Setting root UID and GID also seems to be necessary, otherwise `make Image`
+# starts asking interactive questions like that one:
+# User ID to map to 0 (user root) (INITRAMFS_ROOT_UID) [0] (NEW)
+tee "$fragment" << EOF
+CONFIG_INITRAMFS_SOURCE="$ROOTFS"
+EOF
+
+KCONFIG_CONFIG="$KERNEL_CONFIG" "$KERNEL_DIR/scripts/kconfig/merge_config.sh" "$KERNEL_CONFIG" "$fragment"
+
+rm "$fragment"
```

Comment:
nit: I suggest doing this in `trap "rm $fragment"` EXIT or something like that to avoid with accidental left over - but that's a bit too picky. I'm OK with it if it stays and you don't feel like pushing a new PR with this improvement :)

Reply:
Interestingly it is hard to get rid of a race condition. The best we can do without knowing the file name before creating it is:
```
fragment=$(mktemp)
trap "rm "$fragment"" EXIT
```
If we interrupt the script at the right moment, the file won't be cleaned up. If we create the filename without creating the file, we introduce another race condition where the same name could be used by 2 different processes. Python's `tempfile.NamedTemporaryFile` also seems to be affected by this kind of race condition between creation and cleanup setup.